### PR TITLE
chore: integrate Kotlin logging

### LINKED ISSUE #400: Integrate Kotlin logging with FTL
There are a couple of issues:

- [ ] We're using Logback, and it's not configured to write logs anywhere (seemingly).
- [x] gRPC logs spew vast quantities to the console
- [ ] We're not formatting Logback entries in a way that the Runner expects - the logger should write to stderr in JSON format (run the cluster with `LOG_JSON=1 ftl-dev` to see what the structure is), so that the Runner can parse and forward the logs on to the Controller.

### PR DESCRIPTION
**Pull request by worstell**
fixes #400

### CODE REVIEW

File: kotlin-runtime/ftl-runtime/src/main/kotlin/xyz/block/ftl/logging/Logging.kt [hunk 1]
```diff
@@ -29,18 +36,30 @@ class Logging {
...
 
-      logging.appender.encoder = ple
+      logging.appender.target = ConsoleTarget.SystemErr.toString()
       logging.appender.context = logging.lc
+      logging.appender.encoder = je
       logging.appender.start()
 
-      logger(Logger.ROOT_LOGGER_NAME).level = Level.INFO
+      val rootLogger = logger(Logger.ROOT_LOGGER_NAME)
+      val rootLevelCfg = Level.valueOf(System.getenv("LOG_LEVEL") ?: DEFAULT_LOG_LEVEL)
```
Comment (worstell) [inline kotlin-runtime/ftl-runtime/src/main/kotlin/xyz/block/ftl/logging/Logging.kt:61]: kotlin runtime will respect the LOG_LEVEL env variable - if none is provided the root logger defaults to info level at minimum, which suppresses the spammy GRPC debug logs

File: kotlin-runtime/ftl-runtime/src/main/kotlin/xyz/block/ftl/logging/Logging.kt [hunk 2]
```diff
@@ -29,18 +36,30 @@ class Logging {
...
 
-      ple.pattern = "%date %level %logger{10} - %msg%n"
-      ple.context = logging.lc
-      ple.start()
+      val providers: JsonProviders<ILoggingEvent> = je.providers
+      providers.setContext(je.context)
+      // Custom LogLevelJsonProvider converts level value to lowercase
+      providers.addProvider(object : LogLevelJsonProvider() {
+        override fun writeTo(generator: JsonGenerator, event: ILoggingEvent) {
+          JsonWritingUtils.writeStringField(generator, fieldName, event.level.toString().lowercase())
```
Comment (alecthomas) [inline kotlin-runtime/ftl-runtime/src/main/kotlin/xyz/block/ftl/logging/Logging.kt:48]: How does it know the correct field names to use?

ie. we have

```json
{"message": "X", "error": "Y", "time": "Z", "attributes": {"..."}}
```

**Review by alecthomas — APPROVED** [review]
Awesome!

File: kotlin-runtime/ftl-runtime/src/main/kotlin/xyz/block/ftl/logging/Logging.kt (hunk 2 above)
Comment (worstell) [reply kotlin-runtime/ftl-runtime/src/main/kotlin/xyz/block/ftl/logging/Logging.kt:48]: the kotlin logger relies on the common log infrastructure to supplement "attributes" and "time" when the log is processed via the runner

for the log entry submitted to stderr - we're using the `JsonProviders` to customize the format. these specify field names "message", "error" and "level" and format their values as needed - so when these logs are written they look like `{"level": "<level", "message": "<message">, "error":"<error>"}` and can be parsed into an `Entry` object (and then modified to include additional fields)

**Pull request merged**